src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -426,59 +435,137 @@ def next_edges_by_connector(
+    The function will then modify the modify the strides of `inner_data` to
+    match the ones of `outer_desc`.
+
+    Args:
+        nsdfg_node: The node in the parent SDFG that contains the NestedSDFG.
+        inner_data: The name of the data descriptor that should be processed
+            inside the NestedSDFG (by construction also a connector name).
+        outer_subset: The subset that describes what part of the outer data is
+            mapped into the NestedSDFG.
+        outer_desc: The data descriptor of the data on the outside.
+        ignore_symbol_mapping: If possible the function will perform the renaming
+            through the `symbol_mapping` of the nested SDFG. If `True` then
+            the function will always perform the renaming.
+
     Todo:
-        - Refactor this function.
-        - Handle the case the stride is used somewhere else.
-        - Handle the case where we have an explicit size 1 dimension in slicing.
+        - Handle explicit dimensions of size 1.
     """
-    # We need to propagate the strides inside the nested SDFG on the global arrays
-    new_strides = tuple(
-        stride
-        for stride, to_map_size in zip(
-            outer_strides,
-            edge_data.subset.size(),
-            strict=True,
-        )
-        if to_map_size != 1
-    )
-    inner_desc = nsdfg_node.sdfg.arrays[inner_data]
+    # We need to compute the new strides. In the following we assume that the
+    #  relative order of the dimension does not change, but some dimensions
+    #  that are present on the outside are not present on the inside. For
+    #  example this happens for the Memlet `a[__i0, 0:__a_size1]`.
+    #  We detect this case by checking if that dimension has size 1.
+    # TODO(phimuell): Handle the case were some additional size 1 dimensions are added.
+    inner_desc: dace_data.Data = nsdfg_node.sdfg.arrays[inner_data]
+    inner_shape = inner_desc.shape
+    inner_strides_init = inner_desc.strides
+
+    # TODO(phimuell): For now this is fine, but it should be possisble to allow it.
     assert not inner_desc.transient
 
-    if isinstance(inner_desc, dace.data.Scalar):
-        assert len(new_strides) == 0
+    outer_strides = outer_desc.strides
+    outer_inflow = outer_subset.size()
+
+    new_strides: list = []
+    for dim_ostride, dim_oinflow in zip(outer_strides, outer_inflow, strict=True):
+        current_inner_dim = len(new_strides)
+
+        if inner_shape[current_inner_dim] == 1 and dim_oinflow == 1:
+            # There is an explicit size 1 dimension. Because the only valid
+            #  index for this dimension is `0` we can use any value here.
+            #  To give the compiler more information we explicitly use `0`,
+            #  instead of the outer value.
+            new_strides.append(0)
+
+        elif dim_oinflow == 1:
+            # Only something flows in, thus there is no stride in this dimension.
```

Comment:
I do not understand the comment. Suggestion:
```suggestion
            # This is the case of implicit slicing along one dimension. The inner array descriptor has shape != 1 in `current_inner_dim`, which has to map to a subsequent dimension of `outer_inflow`.
```